src/diffusers/loaders.py

```diff
@@ -1885,16 +1885,18 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
             raise ValueError(f"Unhandled pipeline class: {pipeline_name}")
 
         # remove huggingface url
+        is_URL = False
         for prefix in ["https://huggingface.co/", "huggingface.co/", "hf.co/", "https://hf.co/"]:
             if pretrained_model_link_or_path.startswith(prefix):
                 pretrained_model_link_or_path = pretrained_model_link_or_path[len(prefix) :]
+                is_URL = True
 
         # Code based on diffusers.pipelines.pipeline_utils.DiffusionPipeline.from_pretrained
         ckpt_path = Path(pretrained_model_link_or_path)
         if not ckpt_path.is_file():
             # get repo_id and (potentially nested) file path of ckpt in repo
-            repo_id = os.path.join(*ckpt_path.parts[:2])
-            file_path = os.path.join(*ckpt_path.parts[2:])
+            repo_id = "/".join(ckpt_path.parts[:2]) if is_URL else os.path.join(*ckpt_path.parts[:2])
+            file_path = "/".join(ckpt_path.parts[2:]) if is_URL else os.path.join(*ckpt_path.parts[:2])
```

Comment:
```suggestion
            file_path = "/".join(ckpt_path.parts[2:]) if is_URL else os.path.join(*ckpt_path.parts[2:])
```

Reply:
@Mystfit this should be corrected no? Apart from this this PR looks good to be merged to me :-)

Reply:
@Mystfit up

Reply:
@patrickvonplaten as you're the code owner, you should merge that suggestion because, obviously, it's a typo.

Reply:
No, u didn't, look carefully at else part on 1908: it should start from 2 and to the end. 
```suggestion
            file_path = "/".join(ckpt_path.parts[2:]) if is_URL else os.path.join(*ckpt_path.parts[2:])
```

Reply:
> No, u didn't, look carefully at else part on 1908: it should start from 2 and to the end.

I've just realised something with the latter part of that line. If we're not dealing with a URL and are using a file, then why are we slicing the path from the second position onwards? Wouldn't this make no sense for a file path?